managers/BalanceManager.py

```diff
@@ -2,134 +2,149 @@
-        return False
+        if amount is not None:
+            logger.info("checking to see if amount %s is greater than trade threshold %s", amount, CondexConfig.BITTREX_MIN_BTC_TRADE_AMOUNT)
 
-    def handle_coin(self, coin_above_threshold, coin_below_threshold, percentage_btc_amount,
-                    celery_app):
-        """Handle buying and selling these coins."""
-        logger.debug("Handling sell/buy for %s and %s", coin_above_threshold, coin_below_threshold)
-        coin_balance = DatabaseManager.get_coin_balance_model(coin_above_threshold)
+            over_threshold = float(amount) >= float(CondexConfig.BITTREX_MIN_BTC_TRADE_AMOUNT)
 
-        amounts = self.calculate_amounts(coin_above_threshold, coin_below_threshold,
-                                         percentage_btc_amount)
-        amount_to_sell = amounts["rebalance"]
-        amount_to_buy = amounts["eligible"]
 
-        if coin_balance.TotalCoins >= amount_to_sell:
+            if over_threshold is True:
+                if is_over is False:
+                    logger.info("checking to see if %s is available in BTC", amount)
+                    balance_available = 0.0
+                    btc_balance = DatabaseManager.get_coin_balance_model("BTC")
+                    btc_indexed_coin = DatabaseManager.get_index_coin_model("BTC")
 
-            if self.check_markets(coin_above_threshold, coin_below_threshold):
+                    btc_off = btc_indexed_coin.get_percent_from_coin_target(btc_balance, index_info.TotalBTCVal)
+                    if btc_off <= 0:
+                        return None
 
-                DatabaseManager.create_coin_lock_model(coin_above_threshold)
+                    balance_available = round(btc_balance.BTCBalance * (btc_off / 100), 8)
+                    logger.info("Available BTC balance %s", balance_available)
+                    if balance_available >= amount:
+                        return amount
 
-                DatabaseManager.create_coin_lock_model(coin_below_threshold)
+                    #See if 1x the threshold is available
+                    single_threshold_amount = round(amount / (index_info.BalanceThreshold/100), 8)
 
-                logger.info("Performing Rebalance %s %s - %s %s", coin_above_threshold.upper(),
-                            amount_to_sell, coin_below_threshold.upper(),
-                            amount_to_buy)
-                celery_app.send_task(
-                    'Tasks.perform_rebalance_task', args=[
-                        coin_above_threshold.upper(), amount_to_sell,
-                        coin_below_threshold.upper(), amount_to_buy])
-        else:
-            logger.error("Failed to sell coins - we do not have enough of %s", coin_above_threshold)
+                    if not single_threshold_amount >= em.get_min_buy_btc(pair_string):
+                        single_threshold_amount = em.get_min_buy_btc(pair_string)

```

Comment:
Either I never put in a 2x the min buy here or it got dropped. Either way this is why it's messing up. Either change this to *2 this amount or put in another check to see if the value is >= the minimum trade and use the minimum trade value if it's lower.